### PR TITLE
Fix Swagger Authorization and Client Creation

### PR DESCRIPTION
**Pull request by alihdev**
**Issue 1: URL Misconfiguration**

Encountered an issue with Swagger's integration with Keycloak due to misconfigured authorization URLs:

- **Original** Authorization URL: http://localhost:8080/realms/master/connect/authorize
- **Original** Token URL: http://localhost:8080/realms/master/connect/token

**Resolved** the issue by updating the authentication URLs in the **SwaggerConfigurationHelper** class within **AddAbpSwaggerGenWithOAuth**:

- **Updated** Authorization URL: http://localhost:8080/realms/master/protocol/openid-connect/auth
- **Updated** Token URL: http://localhost:8080/realms/master/protocol/openid-connect/token

Code changes:

```
context.Services.AddAbpSwaggerGenWithOAuth(
    authority: authority,
    scopes: scopes,
    options =>
    {
        options.SwaggerDoc(apiName, new OpenApiInfo { Title = apiTitle, Version = apiVersion });
        options.DocInclusionPredicate((docName, description) => true);
        options.CustomSchemaIds(type => type.FullName);
    },
    authorizationEndpoint: "/protocol/openid-connect/auth",
    tokenEndpoint: "/protocol/openid-connect/token"
);
```

**Issue 2: Missing Client and Scopes**

Encountered errors due to missing client configurations and scopes for **WebGateway_Swagger** & **SwaggerClient**  in **KeyCloakDataSeeder** , resulting in:

- Invalid Client: **WebGateway_Swagger** (Invalid parameter: redirect_uri)
- Invalid Scopes: **SwaggerClient** & **WebGateway_Swagger** (Auth Error[invalid_scope]: Invalid+scopes:+AdministrationService..)

**Resolved by:**

- Refactoring the **CreateSwaggerClientAsync** function.
- Adding the **WebGateway_Swagger** client using **CreateSwaggerClientAsync**.
- Adding scopes in **CreateSwaggerClientAsync** for **SwaggerClient** & **WebGateway_Swagger**.

**Testing Done**
- Tested and verified the corrected URLs with Keycloak authentication for successful authorization and token retrieval.
- Tested the added client configurations and scopes to ensure proper functionality.

**Additional Notes**
- No additional changes made to existing functionalities.

### CODE REVIEW

**Comment by gterdem** [comment]
I think this PR is no longer necessery, it's already being done.

The problem however is shifted to the custom header CORS problem for the keycloak:

image

This is set for the antiforgery security by the Abp.Swashbucke package: https://github.com/abpframework/abp/blob/ce1235a4034f05cfd8bb8add49db529f7ac23c8c/framework/src/Volo.Abp.Swashbuckle/wwwroot/swagger/ui/abp.js#L87-L95

However currently, it is not possible to allow custom headers for CORS on Keycloak server: https://github.com/keycloak/keycloak/issues/12682

I'll investigate more.

File: shared/EShopOnAbp.Keycloak.DbMigrator/KeycloakDataSeeder.cs [hunk 1]
```diff
@@ -329,6 +330,16 @@ private async Task CreateSwaggerClientAsync()
...
+
+            // Add Scopes For ClientId
+            await AddOptionalClientScopesAsync(
+                clientId,
+                new List<string>
+                {
+                    "AdministrationService", "IdentityService", "BasketService", "CatalogService",
+                    "OrderingService", "PaymentService", "CmskitService"
+                }
+            );
```
Comment (alihdev) [inline shared/EShopOnAbp.Keycloak.DbMigrator/KeycloakDataSeeder.cs:342]: @gterdem 
Add Scopes For ClientId (SwaggerClient & WebGateway_Swagger)

Comment (gterdem) [reply shared/EShopOnAbp.Keycloak.DbMigrator/KeycloakDataSeeder.cs:342]: This is fixed with https://github.com/abpframework/eShopOnAbp/pull/238/commits/704fd816e35d3d7d4c22dc4d0e13fa15eae4acec

File: shared/EShopOnAbp.Shared.Hosting.AspNetCore/SwaggerConfigurationHelper.cs [hunk 2]
```diff
@@ -36,6 +36,9 @@ string apiTitle
                 options.SwaggerDoc(apiName, new OpenApiInfo { Title = apiTitle, Version = apiVersion });
                 options.DocInclusionPredicate((docName, description) => true);
                 options.CustomSchemaIds(type => type.FullName);
-            });
+            },
+            authorizationEndpoint: "/protocol/openid-connect/auth",
+            tokenEndpoint: "/protocol/openid-connect/token"
+        );
```
[the remark this answers is not in the document]
Comment (gterdem) [reply shared/EShopOnAbp.Shared.Hosting.AspNetCore/SwaggerConfigurationHelper.cs:42]: This is fixed via https://github.com/silentpartnersoftware/Keycloak.Net/pull/5

**Pull request closed without merging**